pkg/api/validation/validation.go

```diff
@@ -1022,11 +1022,19 @@ func ValidatePersistentVolumeClaim(pvc *api.PersistentVolumeClaim) field.ErrorLi
 func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *api.PersistentVolumeClaim) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newPvc.ObjectMeta, &oldPvc.ObjectMeta, field.NewPath("metadata"))
 	allErrs = append(allErrs, ValidatePersistentVolumeClaim(newPvc)...)
-	// if a pvc had a bound volume, we should not allow updates to resources or access modes
-	if len(oldPvc.Spec.VolumeName) != 0 {
+	// PVController needs to update PVC.Spec w/ VolumeName. Any other change is not allowed.
+	// Claims are immutable in order to enforce quota, range limits, etc. without gaming the system.
+	if len(oldPvc.Spec.VolumeName) == 0 {
+		// Only volumeName changes are allowed. Reset back to empty string after equality check
+		oldPvc.Spec.VolumeName = newPvc.Spec.VolumeName
 		if !api.Semantic.DeepEqual(newPvc.Spec, oldPvc.Spec) {
-			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "spec is immutable once a claim has been bound to a volume"))
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "spec is immutable after creation"))
 		}
+		oldPvc.Spec.VolumeName = ""
+	}
+	// after a pvc is bound to a volume, no changes are allowed
+	if len(oldPvc.Spec.VolumeName) != 0 {
```

Comment:
I understand what you're saying and I believe that's what this code is doing.  The first allows only volumeName update _once_ and then all subsequent updates are rejected outright. Previously, this `if` had its own equality check. I removed it for outright rejection. Is this not correct?

As for your second point, that's what @derekwaynecarr is doing in his PR. Quota at bind time (with the example 10g volume) would have been much more difficult, so he's applying quota with what was requested (1g), which is what you said in your example.